### DISCUSSION TITLE
v6.0.0

### DISCUSSION THREAD

**Discussion by glenn2223** [glenn2223/vscode-live-sass-compiler#263] [Announcements]
# 6.0.0 - 2023-02-07

<small>[Compare to previous release][comp:6.0.0]</small>

### Breaking changes

-   The following deprecated settings have been removed:
    -   `liveSassCompile.settings.formats[].savePathSegmentKeys`
    -   `liveSassCompile.settings.formats[].savePathReplaceSegmentsWith`
    -   _**Note:** other deprecated settings will remain in place to maintain backwards compatibility, or until SASS2.0_
-   Requires VS Code v1.74 or later

### Added

-   New `liveSassCompile.settings.useNewCompiler` setting. **Edit:** "The new API is designed to be more idiomatic, performant, and usable than the old one"  - go on, kick the tyres on this new entrypoint  
~Use it to try the SASS 2.0 compiler - before it becomes mandatory~  [see comment](https://github.com/glenn2223/vscode-live-sass-compiler/discussions/263#discussioncomment-4918137)

### Fixed

-   `debugFileList` command now correctly includes the files that fall under the excluded and partial patterns
-   All node module paths are now resolved correctly (when utilising the `~/[node module name]/file_path` feature)
-   Any output (`Information` or higher) is saved in the output window. **It still only pops up depending on your setting.**  
_This is to match the original extension._

### Updated

-   `autoprefixer` to `10.4.13` [Changelog][cl:ap]
-   `fdir` to `5.3.0` [Changelog][cl:fd]
-   `postcss` to `8.4.21` [Changelog][cl:pc]
-   `sass` to `1.58.0` [Changelog][cl:sa]
-   Various dev dependency updates _(nothing user facing)_

[comp:6.0.0]: https://github.com/glenn2223/vscode-live-sass-compiler/compare/v5.5.1...v6.0.0
[cl:ap]: https://github.com/postcss/autoprefixer/blob/main/CHANGELOG.md
[cl:fd]: https://github.com/thecodrr/fdir/releases
[cl:pc]: https://github.com/postcss/postcss/blob/main/CHANGELOG.md
[cl:sa]: https://github.com/sass/dart-sass/blob/main/CHANGELOG.md

<hr /><em>This discussion was created from the release <a href='https://github.com/glenn2223/vscode-live-sass-compiler/releases/tag/v6.0.0'>v6.0.0</a>.</em>

**Comment by glenn2223** [comment]
New major release, hope you all like it... still 😉

**Reply by SammyAbukmeil** [reply]
Love your plugin! (we use it as part of our curriculum at a software engineering bootcamp for our pre-react content)

Am I right in saying the SASS 2.0 compiler is Dart Sass 2.0.0? [Dart Sass - GitHub Milestone 2.0.0](https://github.com/sass/dart-sass/milestone/3)

Is there a concise list of changes anywhere?

**Reply by glenn2223** [reply]
> Am I right in saying the SASS 2.0 compiler is Dart Sass 2.0.0? [Dart Sass - GitHub Milestone 2.0.0](https://github.com/sass/dart-sass/milestone/3)
> 
> Is there a concise list of changes anywhere?

I think I may have been a bit presumptive in my statement here.

I tested it locally and saw that division operators were operating to SASS 2.0 spec. However, other breaking changes are still being flagged and not throwing errors. I have a screenshot below

I'm pretty sure it's an oversight/issue rather than anything else. @nex3 , as you worked on the spec, can you please confirm?

![image](https://user-images.githubusercontent.com/33450392/217297636-74a3fded-ade3-43fb-9f98-e041f0746e48.png)

---

> Love your plugin! (we use it as part of our curriculum at a software engineering bootcamp for our pre-react content)

Glad you like it! And that so many people will be getting hands-on with it in your bootcamp

**Reply by glenn2223** [reply]
So, I'm certain this is an overstatement.

I will redact this statement but keep the setting in place. As the old compiler callback is technically being deprecated, using this new setting allows you to get hands-on with the new compiler entry point. Not only will this entry point eventually run per SASS 2.0 spec (which seems to be planned for 2024), but it will also allow you to do some compiler checks on the redesign that SASS has done (like checking for errors - though I'm sure there wouldn't be many; if any at all)
